### PR TITLE
feat(install): run config validation at the beginning of the install command

### PR DESCRIPTION
**Pull request by ctrombley**
...plus some changes to `scenarioBuilder.go` to get the `installTest` command working again

### Testing

```
HTTPS_PROXY=localhost:1234 newrelic install
```

output:
```
10:48 $ HTTPS_PROXY=localhost:1234 ./bin/darwin/newrelic install

   _   _                 ____      _ _
  | \ | | _____      __ |  _ \ ___| (_) ___
  |  \| |/ _ \ \ /\ / / | |_) / _ | | |/ __|
  | |\  |  __/\ V  V /  |  _ |  __| | | (__
  |_| \_|\___| \_/\_/   |_| \_\___|_|_|\___|

  Welcome to New Relic. Let's install some instrumentation.

  Questions? Read more about our installation process at
  https://docs.newrelic.com/


INFO Validating connectivity to the New Relic platform... 
ERROR POST https://api.newrelic.com/graphql giving up after 4 attempt(s): Post "https://api.newrelic.com/graphql": proxyconnect tcp: dial tcp [::1]:1234: connect: connection refused 
FATAL We encountered an error during the installation: There was an error connecting to New Relic platform. This is an indication that your firewall or proxy settings do not allow outbound traffic to the New Relic platform. To configure the use of an HTTP proxy, use the HTTP_PROXY and/or HTTPS_PROXY environment variables. For more details visit https://github.com/newrelic/newrelic-cli/blob/main/docs/GETTING_STARTED.md#using-an-http-proxy. If this problem persists please visit the documentation and support page for additional help here: https://one.nr/06vjAeZLKjP

### CODE REVIEW

**Review by jbeveland27 — DISMISSED** [review]
Looks good! There's just some linting issues in `internal/diagnose/errors.go`

File: internal/diagnose/errors.go [hunk 1]
```diff
@@ -2,77 +2,11 @@ package diagnose
...
-func (e ErrPostEvent) Unwrap() error {
-	return e.innerErr
-}
-
-var ErrLicenseKey = errors.New("")
-var ErrInsightsInsertKey = errors.New("")
+var (
+	ErrConnection        = errors.New("there was an error connecting to New Relic platform. This is an indication that your firewall or proxy settings do not allow outbound traffic to the New Relic platform. To configure the use of an HTTP proxy, use the HTTP_PROXY and/or HTTPS_PROXY environment variables. For more details visit https://github.com/newrelic/newrelic-cli/blob/main/docs/GETTING_STARTED.md#using-an-http-proxy")
+	ErrValidation        = errors.New("there was a failure locating the data that was posted to New Relic. Please try again later or contact New Relic support. For real-time platform status info visit https://status.newrelic.com/")
+	ErrDiscovery         = errors.New("failed to detect your system's hostname. Please contact New Relic support.")
```
Comment (jbeveland27) [inline internal/diagnose/errors.go:8]: Linter doesn't like the punctuation here
image

Comment (ctrombley) [reply internal/diagnose/errors.go:9]: fixed

**Review by jbeveland27 — APPROVED** [review]
🐑

**Pull request merged**